File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/extras/widgets/ImageBackdrop.java
package amagi82.modularcharactersheetcreator.ui.extras.widgets;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.widget.ImageView;

/*
    Convenience class to measure an ImageView for the aspect ratio of the Game System splash screens
 */
public class ImageBackdrop extends ImageView {

    public ImageBackdrop(Context context) {
        super(context);
    }

    public ImageBackdrop(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ImageBackdrop(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP) public ImageBackdrop(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int width = MeasureSpec.getSize(widthMeasureSpec);
        int height = width * 154 / 305;
        setMeasuredDimension(width, height);
    }
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/_extras/widgets/ImageBackdrop.kt
package amagi82.modularcharactersheetcreator.ui._extras.widgets

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet
import android.view.View

/*
    Convenience class to measure an ImageView for the aspect ratio of the Game System splash screens
 */
class ImageBackdrop(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : AppCompatImageView(context, attrs, defStyleAttr) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val width = View.MeasureSpec.getSize(widthMeasureSpec)
        val height = width * 154 / 305
        setMeasuredDimension(width, height)
    }
}
